### DISCUSSION TITLE
How to receive current value?

### DISCUSSION THREAD

**Discussion by davfl** [Budibase/budibase#13054] [Help]
when I apply in the days filed the "on change", I can't get the current enddate but it gives me a null object, how can I solve it?

![Screenshot 2024-02-16 124338](https://github.com/Budibase/budibase/assets/100681729/079d8164-8570-4e0d-8c8e-f2e6833c1b79)

**Comment by ConorWebb96** [comment]
Hey @davfl,

Click on **Days** in your screenshot then use the `field value` binding.

**Reply by davfl** [reply]
<img width="197" alt="1" src="https://github.com/Budibase/budibase/assets/100681729/c6bdaf1f-0458-437b-a019-8129f19f3caa">

this is how my form is set, the user should select first the start date and then check the box if he wants to choose the end date using the time passed in the three input box. After setting the time passed it should automatically change the value in date end (date start + the time passed) and in minutes quantity (transform the 3 values in minutes like 1 day it should be 1440 minutes). 

in the number field days is programmed two onChange functions that change the values in date end and in minutes quantity. In the screenshot below shows the function for the date end change 
![2](https://github.com/Budibase/budibase/assets/100681729/663a304f-c34e-40f1-a505-9157d61f2fef)

the result is this:
<img width="192" alt="4" src="https://github.com/Budibase/budibase/assets/100681729/32a687df-07ff-416d-8ced-6d9f7cf67947">


as you see the minutes quantity doesn't change, how can i fix it?

**Reply by davfl** [reply]
The screenshot below shows the function of the second onclick:
<img width="347" alt="5" src="https://github.com/Budibase/budibase/assets/100681729/940d37ed-6909-4ff9-9fe0-89606fdcd096">

**Comment by ConorWebb96** [comment]
Question was asked via email support. Summary of the answered provided is given below.

From looking at the discussion I believe you want to fill in the 3 fields based on the date field's value thats seen above in the screenshot? It looks like your days value is being set but the other fields are being left blank is this the issue you are facing?

The reason that this is only being applied is likely due to your actions. (I can't see what actions you have attached so I'm assuming this.) I can see you only have 2 attached to the on change. You will need to split this JS out. Then add the related JS that will set the value of the hours and minutes fields into their own update field value actions. This will run along with your days update field value action and should apply the values to these fields as well.
